src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -729,6 +729,41 @@ public static void Uri_CombineUsesNewUriString()
             Assert.Equal(Combined, new Uri(baseUri, RelativeUriString).AbsoluteUri);
         }
 
+        [Theory]
+        [InlineData("http://bar/Testue/testImage.jpg")]
+        // Tests that internal Uri info were properly applied during a Combine operation when URI contains non-ascii character.
+        [InlineData("http://bar/Test�/testImage.jpg")]
+        public static void Uri_CombineWithAbsoluteHttpUriResultInAbsoluteHttpSchema(string fileUri)
+        {
+            string baseUriString = "combine-scheme://foo";
+
+            var baseUri = new Uri(baseUriString, UriKind.Absolute);
+            var uri = new Uri(fileUri);
+            var resultUri = new Uri(baseUri, uri);
+            string resultUriString = resultUri.ToString();
+
+            Assert.DoesNotContain(baseUriString, resultUriString);
+            Assert.Contains("http://", resultUriString);
+            Assert.Equal(fileUri, resultUriString);
```

Comment:
These asserts seem more relaxed than we might want.

Can we make the test take exact expected strings as input instead?
E.g. something along the lines of

```c#
void Test(string uriString, string expectedHost, string expectedAbsoluteUri, string expectedUriString)
{
    var baseUri = new Uri("http://foo/bar");
    var uri = new Uri(uriString);
    var resultUri = new Uri(baseUri, uri);

    Assert.Equal(expectedHost, resultUri.IdnHost);
    // ...
}
```